arch/risc-v/src/common/riscv_schedulesigaction.c

```diff
@@ -126,8 +126,6 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
                * been delivered.
                */
 
-              tcb->xcp.saved_regs = (uintptr_t *)CURRENT_REGS;
```

Comment:
@Ouss4, where is duplication?

Reply:
The call to `riscv_savestate(tcb->xcp.saved_regs);` does the same thing as this line.